src/VirtoCommerce.SitemapsModule.Web/Controllers/Api/SitemapsModuleApiController.cs

```diff
@@ -317,14 +318,33 @@ public async Task<ActionResult<SitemapDownloadNotification>> DownloadSitemap(str
         }
 
         /// <summary>
-        /// 
+        ///
         /// </summary>
         /// <param name="storeId"></param>
         /// <param name="baseUrl"></param>
         /// <param name="notification"></param>
         /// <returns></returns>
         [ApiExplorerSettings(IgnoreApi = true)]
-        public async Task BackgroundDownload(string storeId, string baseUrl, SitemapDownloadNotification notification)
+        public Task BackgroundDownload(string storeId, string baseUrl, SitemapDownloadNotification notification)
+        {
+            // We cannot use storeId.IndexOfAny(Path.GetInvalidFileNameChars()) != -1 to validate path because default
+            // sanitizer for Sonar Cube do not trust it, so we use Regex here with same logic. Check this out
+            // https://community.sonarsource.com/t/help-sonarcloud-with-understanding-the-usage-of-untrusted-and-tainted-input/9873/7
+            // Btw, we cannot move this to extansion or any method from here because sonar ignore any outer checks :(
+            if (!Regex.IsMatch(storeId, "^[a-zA-Z0-9]+$"))
+            {
+                throw new ArgumentException($"Incorrect name of store {storeId}");
+            }
+
+            if (!Uri.TryCreate(baseUrl, UriKind.Absolute, out var correctUri) || correctUri.Segments.Length > 1)
```

Comment:
We shouldn't limit the URLs being only the  root domains

Reply:
So need to remove check on subdirectories